### PR TITLE
adds DDL storage options documentation

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: blackbox/docs/general/ddl/storage.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Column Store
+============
+
+Beside of storing the row data as-is (and indexing each value if wanted), each
+value term is stored into a `Column Store`_ by default. The usage of a `Column
+Store`_ is greatly improving global aggregations and groupings and enables
+ordering possiblity as the data for one column is packed at one place.
+
+Turning the `Column Store`_ off will remove the length limitation of 32766 bytes
```
Comment (mfussenegger) [inline blackbox/docs/general/ddl/storage.txt]: I think here is a bit of a leap since the length limitation isn't mentioned before.
Maybe add a short introduction along the lines of "Using the column storage limits the values of string columns to a maximal length of .."

File: blackbox/docs/general/ddl/storage.txt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+but also will disable ordering functionallity.
```
Comment (mfussenegger) [inline blackbox/docs/general/ddl/storage.txt]: Can't we do a fallback? Making it more expensive but not completely disabling it?

Comment (seut) [reply blackbox/docs/general/ddl/storage.txt]: I think we can do, but I'm not sure if we want to do that right now. I'd prefer changing the documentation after we've implemented the fallback.

File: blackbox/docs/general/ddl/storage.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
+=========
+ Storage
+=========
+
+How data is stored can be tuned on each column similar to how indexing is defined.
```
Comment (matriv) [inline blackbox/docs/general/ddl/storage.txt]: Would rephrase to: "Data storage options can be tuned for each..."

File: blackbox/docs/general/ddl/storage.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+Column Store
+============
+
+Beside of storing the row data as-is (and indexing each value if wanted), each
```
Comment (matriv) [inline blackbox/docs/general/ddl/storage.txt]: if wanted -> by default

File: blackbox/docs/general/ddl/storage.txt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+value term is stored into a `Column Store`_ by default. The usage of a `Column
+Store`_ is greatly improving global aggregations and groupings and enables
+ordering possiblity as the data for one column is packed at one place. Using the
+`Column Store`_ limits the values of :ref:`data-type-string` columns to a
+maximal length of 32766 bytes.
+
+Turning the `Column Store`_ off will remove the length limitation but also will
```
Comment (matriv) [inline blackbox/docs/general/ddl/storage.txt]: Turning off the `Column Store`_ will remove the...

File: blackbox/docs/general/ddl/storage.txt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+::
+
+    cr> CREATE TABLE t1 (
+    ...   id INTEGER,
+    ...   url STRING STORAGE (columnstore = false)
+    ... );
+    CREATE OK, 1 row affected  (... sec)
+
+Doing so will enable support for inserting strings longer than 32766 bytes into
+the ``url`` column, but will also disable the possibility to sort on the ``url``
```
Comment (matriv) [inline blackbox/docs/general/ddl/storage.txt]: Is it true? We still can sort in memory, right?

Comment (seut) [reply blackbox/docs/general/ddl/storage.txt]: as already discussed at previous comment by @mfussenegger , we could. pls read discussion

Comment (matriv) [reply blackbox/docs/general/ddl/storage.txt]: @seut, sorry missed that.

File: blackbox/docs/sql/statements/create_table.txt [hunk 7]
```diff
@@ -31,7 +31,7 @@ Synopsis
 
 where ``base_column_definition``::
 
-    column_name data_type [ column_constraint [ ... ] ]
+    column_name data_type [ column_constraint [ ... ] ]  [ storage_option ]
```
Comment (matriv) [inline blackbox/docs/sql/statements/create_table.txt]: `storage_option` is used here as placeholder for the whole expression and also below for the specific storage param.

I would rename this to something like storage_configuration or storage_option**s** , or the other one (https://github.com/crate/crate/pull/6583/commits/112c07cb9c73f01af9894761d49ac0da3754b900#diff-49b1eb6da27aeff61046466a2ac4697aR52) to just `option` or `param`

File: blackbox/docs/sql/statements/create_table.txt [hunk 8]
```diff
@@ -47,6 +47,10 @@ where ``column_constraint`` is::
                             FULLTEXT [ WITH ( analyzer = analyzer_name ) ]  }
     }
 
+where ``storage_option`` is::
+
+    STORAGE WITH ( storage_option = value_expression [, ... ] )
```
Comment (matriv) [inline blackbox/docs/sql/statements/create_table.txt]: shouldn't we point to what the available storage options are? (currently only one of course)

File: blackbox/docs/sql/statements/create_table.txt [hunk 9]
```diff
@@ -130,10 +130,15 @@ Column Constraints
 Column constraints are constraints that are applied on each column of the table
 separately.
 
-The supported column constraints are:
```
Comment (matriv) [inline blackbox/docs/sql/statements/create_table.txt]: You could also fix it separately in master.
